### ISSUE TITLE
Feature Request: Add collectd parser

### ISSUE THREAD

**Issue by gunnaraasen** [influxdata/telegraf#1100]
Should be fairly straightforward to port the existing code from InfluxDB's [collectd input service](https://github.com/influxdata/influxdb/tree/master/services/collectd).

**Comment by sparrc** [comment]
since we already have the tcp_listener and udp_listener plugins, we should be able to just write a parser for the collectd format (https://github.com/influxdata/telegraf/tree/master/plugins/parsers)

**Comment by panda87** [comment]
We workaround it with graphite output from Collectd, and used UDP listener on the host with the latest template feature, and it works very well

**Comment by bfgoodrich** [comment]
I really like the idea of having a specific plugin for collectd

**Comment by danielnelson** [comment]
It won't be a whole input plugin, just a new parser to be used with socket_listener like this:
```
[[inputs.socket_listener]]
  service_address = "udp://127.0.0.1:25826"
  data_format = "collectd"
  name_prefix = "collectd_"
```

**Comment by danielnelson** [comment]
One limitation of this method is we can't support authentication, encryption, signing, or validation against a types database.

**Comment by danielnelson** [comment]
> One limitation of this method is we can't support authentication, encryption, signing, or validation against a types database.

This is not correct, all of these can be supported via additional parser options:
```
  collectd_auth_file = "/etc/collectd/auth_file"
  collectd_security_level = "sign"
  collectd_typesdb = "/usr/share/collectd/types.db"
```

**Comment by danielnelson** [comment]
Implemented in #2654
